servicetests/tests/FleetProvisioning/src/main/java/fleetProvisioning/FleetProvisioning.java

```diff
@@ -0,0 +1,252 @@
+/**
+ * Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0.
+ */
+
+package fleetProvisioning;
+
+import software.amazon.awssdk.crt.CRT;
+import software.amazon.awssdk.crt.CrtResource;
+import software.amazon.awssdk.crt.CrtRuntimeException;
+import software.amazon.awssdk.crt.mqtt.MqttClientConnection;
+import software.amazon.awssdk.crt.mqtt.MqttClientConnectionEvents;
+import software.amazon.awssdk.crt.mqtt.QualityOfService;
+import software.amazon.awssdk.crt.mqtt5.Mqtt5ClientOptions;
+import software.amazon.awssdk.crt.mqtt5.Mqtt5Client;
+import software.amazon.awssdk.crt.mqtt5.packets.ConnectPacket;
+import software.amazon.awssdk.iot.AwsIotMqttConnectionBuilder;
+import software.amazon.awssdk.iot.AwsIotMqtt5ClientBuilder;
+import software.amazon.awssdk.iot.iotidentity.IotIdentityClient;
+import software.amazon.awssdk.iot.iotidentity.model.CreateKeysAndCertificateRequest;
+import software.amazon.awssdk.iot.iotidentity.model.CreateKeysAndCertificateResponse;
+import software.amazon.awssdk.iot.iotidentity.model.CreateKeysAndCertificateSubscriptionRequest;
+import software.amazon.awssdk.iot.iotidentity.model.ErrorResponse;
+import software.amazon.awssdk.iot.iotidentity.model.RegisterThingRequest;
+import software.amazon.awssdk.iot.iotidentity.model.RegisterThingResponse;
+import software.amazon.awssdk.iot.iotidentity.model.RegisterThingSubscriptionRequest;
+
+import java.nio.file.Files;
+import java.nio.file.Paths;
+
+import java.util.HashMap;
+import com.google.gson.Gson;
+import java.util.concurrent.CompletableFuture;
+import java.util.concurrent.TimeUnit;
+
+import utils.commandlineutils.CommandLineUtils;
+import utils.mqttclientconnectionwrapper.*;
+import ServiceTestLifecycleEvents.ServiceTestLifecycleEvents;
+
+public class FleetProvisioning {
+
+    static CompletableFuture<Void> gotResponse;
+    static IotIdentityClient iotIdentityClient;
+
+    static CreateKeysAndCertificateResponse createKeysAndCertificateResponse = null;
+    static RegisterThingResponse registerThingResponse = null;
+
+    static long responseWaitTimeMs = 5000L; // 5 seconds
+
+    static MqttClientConnectionWrapper createConnection(CommandLineUtils.SampleCommandLineData cmdData, Integer mqttVersion) {
+        if (mqttVersion == 3) {
+            try (AwsIotMqttConnectionBuilder builder = AwsIotMqttConnectionBuilder
+                    .newMtlsBuilderFromPath(cmdData.input_cert, cmdData.input_key)) {
```

Comment:
Is it possible to move the client creation to ConnectionWrapper so that we could reuse for other service client.?

Reply:
I was going to do this in the next PR, but can do it here as well